packages/cli/src/services/check-runner.ts

```diff
@@ -104,19 +104,19 @@ export default class CheckRunner extends EventEmitter {
       checkRun.group = this.groups[check.groupId!.ref].synthesize()
       delete checkRun.groupId
     }
-    this.emit(Events.CHECK_REGISTERED, checkRun)
+    this.emit(Events.CHECK_REGISTERED, check)
     try {
       await checksApi.run(checkRun)
+      return check
     } catch (err: any) {
       if (err?.response?.status === 402) {
         const errorMessage = `Failed to run a check. ${err.response.data?.message}`
-        this.emit(Events.CHECK_FAILED, checkRun, new Error(errorMessage))
-        this.emit(Events.CHECK_FINISHED, check)
-        // TODO: Find a way to abort. The latest version supports this but doesn't work with TS
-        return
+        this.emit(Events.CHECK_FAILED, check, new Error(errorMessage))
+      } else {
+        this.emit(Events.CHECK_FAILED, check, err)
       }
-      this.emit(Events.CHECK_FAILED, checkRun, err)
       this.emit(Events.CHECK_FINISHED, check)
+      throw new Error('Failed to run a check.')
```

Comment:
The catch block should handle the api issues. I actually don't get why we need to throw another error here

Reply:
@umutuzgur , we weren't throwing the error when `status !== 402`. It's important to differentiate the `fulfilled` from the `rejected` promise.

Reply:
I'm also confused here. Since we catch the error, shouldn't the promise from `scheduleCheck()` always be fulfilled? I don't see how the promise would become stuck in a pending state.

Reply:
@umutuzgur , @clample , I updated the PR comments explaining the solution. Let me know if you need more details.

Reply:
@nahuelon thanks for clarifying. Nice find with the `getSourceFile()` issue. 

I'm not sure what you and @umutuzgur think, but I think that it would still be worth looking into this bug further. The current PR seems to fix it, but I think it's worth making sure we understand 100% what happened. The main open questions for me are why the `runner.run()` promise isn't being fulfilled when all checks are finished, and why  `scheduleCheck` wasn't being fulfilled before.

Let me know if you'd like to pair on this. I'm happy to help take a look.